### DISCUSSION TITLE
config error: matchDepNames

### DISCUSSION THREAD

**Discussion by nejec** [renovatebot/renovate#19811] [Request Help] [answered]
Hey, 

I am running renovate to update dependencies mainly for `gitlabci`/`gitlabci-include` managers, however I don't set the used managers if something other needs to be updated.

However, I am having issues with renovate configuration, where I am constantly getting the following error `Invalid configuration option: packageRules[4].matchDepNames` (full debug error at the end).

The error occurs, when I try to exclude updating `.gitlab-ci.yml` definition of `image: python:3.9` to renovate proposed `image: python:3.11`. For that reason, I have created a package rule, which somehow is not processed right (or I am doing something wrong.

```json
    {
      "matchDepNames": ["python"],
      "matchUpdateTypes": ["major", "minor"],
      "enabled": false
    },
```

I have tried setting it up to `matchPackageName: ["python"]` but it didn't work and `python:3.9` image reference in `.gitlab-ci.yml` was upgraded to `3.11`.

I am loosing ideas, what am I doing wrong. Any ideas? Is this a bug or should I use some different approach?

Below is my current configuration

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base", ":disableDependencyDashboard", ":rebaseStalePrs"],
  "commitMessagePrefix": "deps: ",
  "ignorePresets": [":prHourlyLimit2"],
  "labels": ["renovate"],
  "pruneStaleBranches": true,
  "recreateClosed": false,
  "separateMajorMinor": true,
  "rangeStrategy": "pin",
  "docker": {"pinDigests": true},
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": false,
      "automergeType": "branch"
    },
    {
      "matchUpdateTypes": ["major", "minor", "patch"],
      "matchManagers": ["gitlabci-include"],
      "labels": ["renovate", "gitlab-ci-templates"],
      "groupName": "gitlab-ci template dependencies"
    },
    {
      "matchUpdateTypes": ["major", "minor", "patch", "digest"],
      "matchManagers": ["gitlabci"],
      "labels": ["renovate", "gitlab-ci"],
      "groupName": "gitlab-ci dependencies"
    },
    {
      "matchUpdateTypes": ["major"],
      "matchManagers": [
        "npm",
        "pip_requirements",
        "dockerfile",
        "docker-compose"
      ],
      "enabled": false
    },
    {
      "matchDepNames": ["python"],
      "matchUpdateTypes": ["major", "minor"],
      "enabled": false
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "matchManagers": ["pip_requirements"],
      "labels": ["pip"],
      "groupName": "python pip dependencies"
    }
  ]
}
```

full error

```
 INFO: Repository has invalid config (repository=polariumrnd/digital/common/pd-application-logging)
       "error": {
         "validationSource": ".renovaterc.json",
         "validationError": "The renovate configuration file contains some invalid settings",
         "validationMessage": "Invalid configuration option: packageRules[4].matchDepNames",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at mergeRenovateConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/merge.ts:197:19)\n    at getRepoConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/config.ts:12:12)\n    at initRepo (/usr/src/app/node_modules/renovate/lib/workers/repository/init/index.ts:38:12)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:47:14)\n    at attributes.repository (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:173:11)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:158:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:17:22"
       }
```

**Comment by nejec** [comment]
If I use the same configuration in other repository, the error is not there, so now I trying to find out what are the differences.

I am attaching full debug log with error...
[renovate.log](https://github.com/renovatebot/renovate/files/10403774/renovate.log)

really, no clue what am I doing wrong...

Thank you in advance

**Comment by nejec** [comment]
As it seems, this bug (or feature) was updated with newer releases of `renovate`:

This issues exists in `renovate:34.70.4`, while `renovate:34.84.2` fixes the issue.

**Comment by rarkins** [comment, marked as answer]
`matchDepNames` wasn't introduced until v34.74.0: https://github.com/renovatebot/renovate/pull/19565#issuecomment-1365358521

You were using an old version of Renovate, so naturally it didn't know about that. No "issue" existed. No "fix" occurred.
